server/routes.js

```diff
@@ -0,0 +1,98 @@
+const MachineController = require('./controllers/MachineController');
+
+module.exports = (app, redis) => {
+	/***********
+	 * Caching *
+	 ***********/
+	/*
+	 * These methods take the responsibility of caching from the
+	 * controllers and gives it to the router. How it works is
+	 * detailed in the documentation below.
+	 */
+	/**
+	 * ( ͡° ͜ʖ ͡°)
+	 */
+	function setHeaders(req, res, next) {
+		res.setHeader('X-SigApp', 'https://www.youtube.com/watch?v=dQw4w9WgXcQ'); // TODO Make a shortlink
+		next();
+	}
+
+	/**
+	 * The first method in a route declaration is executed first. checkCache checks to make sure this route
+	 * hasn't been requested in the last minute. If it has, it sends the result, and the controller never knew it got a
+	 * request. If it hasn't, it calls next to execute the request as normal.
+	 */
+	async function checkCache(req, res, next) {
+		// TODO time tracking
+
+		let exists = await redis.exists(req.path);
```

Comment:
Why are we messing with the cache at the parsing layer and again at the routing layer? This seems redundant and feels like one case would never be hit. 

I would recommend keeping it at just the data and parsing layer. The upper layers don't need to know. 

Unless I am not understanding the purpose.

Reply:
good point. it is rather redundant